### PR TITLE
fix(opentelemetry): change sample to use Shutdown, and add a unit test

### PR DESCRIPTION
**Pull request by dashpole**
## Description

Fixes b/294608478

Shutdown is preferable to ForceFlush as it performs additional cleanup (e.g. closing connections) that is appropriate for most uses of OpenTelemetry.  ForceFlush is intended for niche use-cases in which the program can be paused, but not shut down, such as serverless environments.

This also updates dependencies.

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] These samples need a new **API enabled** in testing projects to pass: **Cloud Trace API**
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by dashpole** [comment]
cc @tritone

File: opentelemetry/trace/main_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+	// Check that the project contains our trace
+	ctx := context.Background()
+	client, err := trace.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("Failed to create trace client: %v", err)
+	}
+	defer client.Close()
+	// Wait a few seconds to ensure our trace is returned by the trace API
+	time.Sleep(5 * time.Second)
```
Comment (tritone) [inline opentelemetry/trace/main_test.go:56]: I recommend removing this delay and using [`testutil.Retry`](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/internal/testutil/retry.go#L30) to wrap the code below. That way, you can increase the deadline if there is variability in the service availability but less time will be spent waiting in most cases.

File: opentelemetry/trace/main_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func TestWriteTraces(t *testing.T) {
+	// Tests build.
+	m := testutil.BuildMain(t)
+	if !m.Built() {
+		t.Fatalf("failed to build app")
+	}
+	projectID := os.Getenv("GOLANG_SAMPLES_PROJECT_ID")
```
Comment (tritone) [inline opentelemetry/trace/main_test.go:36]: Use [`testutil.SystemTest`](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/internal/testutil/testutil.go#L56) to get the project ID or skip if not available.

**Review by tritone — COMMENTED** [review]
Generally looks good, a few test suggestions.

File: opentelemetry/trace/main_test.go (hunk 1 above)
Comment (dashpole) [reply opentelemetry/trace/main_test.go]: done

File: opentelemetry/trace/main_test.go (hunk 2 above)
Comment (dashpole) [reply opentelemetry/trace/main_test.go]: done

**Review by tritone — APPROVED** [review]

**Pull request merged**